src/mprk.jl

```diff
@@ -379,10 +320,10 @@ function perform_step!(integrator, cache::MPECache, repeat_step = false)
     fill!(P, zero(eltype(P)))
 
     f.p(P, uprev, p, t) # evaluate production terms
-    sum_destruction_terms!(D, P) # store destruction terms in D
     integrator.stats.nf += 1
 
-    build_mprk_matrix!(P, 1, P, D, uprev, dt)
+    build_mprk_matrix!(P, P, uprev, dt)
```

Comment:
This may also be a bit delicate in the future and could require an additional matrix `M`